util/writer.go

```diff
@@ -11,15 +11,16 @@ type CodeWriter interface {
 	Write(code string)
 }
 
-func prefix(indentation int) string {
-	tab := strings.Repeat(" ", 2)
+func prefix(indentation int, numSpacesIndent int) string {
```

Comment:
To make it flexible enough for other languages that want to have 4 spaces for indent.